client/network/src/request_responses.rs

```diff
@@ -163,7 +163,10 @@ pub struct RequestResponsesBehaviour {
 	>,
 
 	/// Pending requests, passed down to a [`RequestResponse`] behaviour, awaiting a reply.
-	pending_requests: HashMap<RequestId, (Instant, oneshot::Sender<Result<Vec<u8>, RequestFailure>>)>,
+	pending_requests: HashMap<
+			(Cow<'static, str>, RequestId),
```

Comment:
It may add a bit of clarity to introduce a custom struct like `ProtocolRequestId`. I generally find it preferable to not use plain tuples in data structures but instead dedicated structs with named fields. Both the name of the type and the names of the fields can help make the code a bit easier to read on first sigh. Just a small suggestion though.

Reply:
Agreed that the above tuple specifically, but oftentimes tuples in general are not descriptive enough to avoid confusion. I find `ProtocolRequestId` a good idea. https://github.com/paritytech/substrate/pull/7957/commits/7ac1b3c777915fee6ead00b201ec916dd2c156d0 introduces the `ProtocolRequestId` struct adding a succinct doc comment. Does the additional struct in combination with the doc comment establish enough clarity here?